### PR TITLE
refactor: allow config to pass extra fields

### PR DESCRIPTION
**Pull request by NotPeopling2day**
### What I did

fixes: #534 

### How I did it

Allow passing of extra fields via the pydantic Config class

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: src/ape_geth/providers.py [hunk 1]
```diff
@@ -118,6 +119,9 @@ class GethNetworkConfig(PluginConfig):
     # Make sure to run via `geth --dev` (or similar)
     local: dict = DEFAULT_SETTINGS.copy()
 
+    class Config:
+        extra = Extra.allow
```
Comment (unparalleled-js) [inline src/ape_geth/providers.py:123]: maybe we need a comment saying what this is for? "Adding ecosystems that use Geth, such as geth-forks"

Comment (unparalleled-js) [reply src/ape_geth/providers.py:123]: also (talked offline) but this is the wrong spot, needs to be in the class below.

After that, things seem to work in ape-fantom!

**Review by unparalleled-js — COMMENTED** [review]
I did more testing!

* I think in that same URI method, we need to make return the default URI if `self.config`  does NOT have attr `self.network.ecosystem.name`.. Otherwise, we always have to specify the values in the config

* Error handling for non-dict extra fields that clearly are not network configs

**Comment by fubuloubu** [comment]
I was able to use this with `ape-fantom` and get some stuff working for me, however I had one bug:
https://github.com/ApeWorX/ape/pull/537/files#diff-006c1d861ec6abe5c7a1135384957b5da1a101f56bb4fb991624988449b9d345R147

changed to `ecosystem_config.get(self.network.name)` because it is `__root__: dict`

**Review by unparalleled-js — APPROVED** [review]
This seems to work!
Also if you combine this with the logic in this PR: https://github.com/ApeWorX/ape/pull/544, the experience is even better

**Review by unparalleled-js — APPROVED** [review]
LGTM

**Pull request merged**